src/__tests__/replaceAroundStep.test.ts

```diff
@@ -10,6 +10,62 @@ import { suggestReplaceAroundStep } from "../replaceAroundStep.js";
 import { type TaggedNode, testBuilders } from "../testing/testBuilders.js";
 
 describe("ReplaceAroundStep", () => {
+  it("should handle setNodeMarkup", () => {
+    const doc = testBuilders.doc(
+      testBuilders.paragraph("first paragraph"),
+      testBuilders.paragraph("second paragraph"),
+    ) as TaggedNode;
+
+    const editorState = EditorState.create({
+      doc,
+      selection: TextSelection.atStart(doc),
+    });
+
+    const originalTransaction = editorState.tr;
+
+    originalTransaction.setNodeMarkup(0, testBuilders.schema.nodes.heading, {
+      level: 2,
+    });
+
+    const step = originalTransaction.steps[0];
+    assert(
+      step instanceof ReplaceAroundStep,
+      "Could not create test ReplaceAroundStep",
+    );
+
+    const trackedTransaction = editorState.tr;
+    suggestReplaceAroundStep(trackedTransaction, editorState, doc, step, [], 1);
+
+    const trackedState = editorState.apply(trackedTransaction);
+
+    const expected = testBuilders.doc(
+      testBuilders.modification(
+        {
+          id: 1,
+          type: "attr",
+          attrName: "level",
+          previousValue: 1, // TBD: previousValue should be 1 (default value), or undefined?
```

Comment:
I _think_ this is fine, though it may prove to be funky for users that attempt to build a UI to display modifications in more detail. I think it's probably ok for now, though